### PR TITLE
feat: distributed authority sample - adding assembly definitions [MTT-8945]

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description
This PR creates assembly definitions for several directories.
Dependencies (for now):
Input -> Unity.InputSystem
Gameplay -> Unity.Netcode.Runtime
Player-> Input, Gameplay, Unity.Netcode.Runtime

### Issue Number(s)
[MTT-8945](https://jira.unity3d.com/browse/MTT-8945)

### Contribution checklist
 - [ N/A] Tests have been added for the project and/or any internal package
 - [ N/A ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Input/AvatarInputs.cs [hunk 1]
```diff
@@ -1,7 +1,9 @@
+using System.Runtime.CompilerServices;
 using UnityEngine;
 using UnityEngine.InputSystem;
 
-namespace com.unity.multiplayer.samples.distributed_authority.input
+[assembly: InternalsVisibleTo("com.unity.multiplayer.samples.socialhub.player")]
```
Comment (RikuTheFuffs) [inline Experimental/DistributedAuthoritySample/Assets/Scripts/Input/AvatarInputs.cs:5]: considering that the AvatarInputs class might go away at any moment in the future, I'd recommend to put the `InternalsVisibleTo` into an AssmeblyInfo.cs file like we do in the Small Scale Template. This centralizes the place where all internals are "shared", and makes it independent from individual script classes: https://github.cds.internal.unity3d.com/unity/com.unity.template.multiplayer.ngo/blob/main/Assets/Scripts/Runtime/AssemblyInfo.cs

**Review by RikuTheFuffs — CHANGES_REQUESTED** [review]

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Input/AvatarInputs.cs (hunk 1 above)
Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/Assets/Scripts/Input/AvatarInputs.cs]: That's neat! Done.

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+namespace Unity.Multiplayer.Samples.SocialHub.Player
 {
     [RequireComponent(typeof(Rigidbody))]
     public class AvatarTransform : NetworkTransform
     {
         [SerializeField]
         Rigidbody m_Rigidbody;
         [SerializeField]
-        PlayerInput m_PlayerInput;
+        MonoBehaviour m_PlayerInput;
```
Comment (RikuTheFuffs) [inline Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs:13]: is this intended? MonoBehaviour could be anything

Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs]: I initially didn't want to include the InputSystem assembly inside the Player's, but yes let's bind them.

**Pull request merged**